docs/source/guides/contributor/chapters/ai_policy.rst

```diff
@@ -0,0 +1,121 @@
+    commit that includes AI-generated code, contributors **must**
+    explicitly state that AI tools were used. This has to be done in the
+    commit message and PR description.
+
+    -   **Example Commit Message/PR Description:**
+
+        ::
+
+            header          <- Limited to 72 characters. No period.
+                            <- Blank line
+            message         <- Any number of lines, limited to 72 characters per line.
+                            <- Blank line
+            Assisted-by:    <- artificial intelligence model, tool, or service which
+                            has been used and how big part of contribution has
+                            been generated (percentage)
+            Reference:      <- External references, one per line (issue, trello, ...)
+            Signed-off-by:  <- Signature and acknowledgment of licensing terms when
+                            contributing to the project (created by git commit -s)
+
+-   **Originality:** If AI was used as a brainstorming or refactoring
+    tool, but the final code is substantially original and reviewed by
+    the contributor, a general disclosure in the PR description is
+    sufficient.
+
+Licensing and Copyright
+-----------------------
+
+Contributors are responsible for ensuring that AI-generated code complies
+with the Avocado's license (GPLv2).
+
+-   **Be Alert to Possible Copyright Issues:** Do not assume that
+    AI-generated code is free of licensing obligations or copyright.
+    Treat it with the same diligence as code copied from other sources.
+    To the extent practical, ensure that AI-generated code does not
+    infringe third-party copyrights.
+-   **Review for Training Data Leakage:** Contributors are expected to
+    enable any code similarity matching or blocking functionality offered
+    by their AI tools if it is available. This helps to identify or
+    prevent potential leakage from proprietary or incompatibly licensed
+    training data. If such instances are identified by the contributor,
+    the contributor must either (a) ensure that the AI-generated code is
+    used in a manner that complies with its license, and ensure that such
+    license is compatible with the license of the project, or (b) the
+    code should be re-written before submission.
+
+Code Quality, Review, and Testing
+----------------------------------
+
+AI-generated code must adhere to the same quality standards as
+human-written code.
+
+-   **Human Review Required:** All AI-generated code, regardless of its
+    source, **must** be thoroughly reviewed by a human contributor. This
+    review should be as rigorous, if not more so, than the review of
+    human-written code.
+-   **Testing:** Comprehensive tests (unit, integration, etc.) must be
+    written for AI-generated code, just as for any other code. This is
+    crucial for verifying its behaviour and preventing regressions.
```

Comment:
What if some coders use AI to write the unit tests too? I assume this is ok but somehow the above statement seems to exclude that possibility.

Reply:
@clebergnu What is the motivation behind the requirement not to generate both tests and implementation at once?

In particular in test-driven development practices, one may even write the test first (which often ends up being more important than the implementation as it defined the user/consumer/caller interface) and ask AI to generate the implementation. So in some imperfect sense the above aligns with this but we could just keep it independent of any choice. So in short - what is the reason for separating tests and implementations in this way and requiring them in certain order or in any order but separately generated?

Reply:
Maybe we can completely remove this, because the initial idea was that contributors needs to understand and test their changes, not just blindly use what was generated. But this is also true for not generated code. So this is one of those parts which @ldoktor was talking about (general recommendations not related to Avocado or AI) and also it's already causing a confusion, therefore I would remove it.